pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -115,6 +121,45 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 	return reconcile.Result{}, nil
 }
 
+func (c *Controller) ensureTerminationGracePeriodExpirationAnnotation(ctx context.Context, node *v1.Node, nodeClaim *v1beta1.NodeClaim) error {
+
+	nodePool, err := nodeclaimutil.NodePoolForNodeClaim(ctx, c.kubeClient, nodeClaim)
+	if err != nil {
+		return fmt.Errorf("ensuring node expiration annotation, %w", err)
+	}
+
+	if nodePool != nil && nodePool.Spec.Disruption.TerminationGracePeriod != nil && nodeClaim.ObjectMeta.DeletionTimestamp != nil {
+		expirationTimeString := nodeClaim.DeletionTimestamp.Time.Add(nodePool.Spec.Disruption.TerminationGracePeriod.Duration).Format(time.RFC3339)
+
+		// set annotation if no annotations exist
+		if node.ObjectMeta.Annotations == nil {
+			return c.annotateTerminationGracePeriodExpirationTime(ctx, node, expirationTimeString)
+		}
```

Comment:
An existence check will work on a nil map so we can drop this statement and just rely on the statement @ line 140 right?